app/notification.go

```diff
@@ -278,16 +265,20 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 			}
 
 			if ShouldSendPushNotification(profileMap[id], channelMemberNotifyPropsMap[id], true, status, post) {
+				mentionType := mentions.Mentions[id]
+
 				replyToThreadType := ""
-				if value, ok := threadMentionedUserIds[id]; ok {
-					replyToThreadType = value
+				if mentionType == ThreadMention {
+					replyToThreadType = model.COMMENTS_NOTIFY_ANY
+				} else if mentionType == CommentMention {
+					replyToThreadType = model.COMMENTS_NOTIFY_ROOT
 				}
 
 				a.sendPushNotification(
 					notification,
 					profileMap[id],
-					mentionedUserIds[id],
-					(channelNotification || hereNotification || allNotification),
+					mentionType == KeywordMention || mentionType == ChannelMention || mentionType == DMMention,
+					mentionType == ChannelMention,
```

Comment:
again, I don't think we should change it now, but feels like sendpushnotification could receive mentiontype instead of multiple comparisons?